apps/examples/testcase/ta_tc/systemio/itc/itc_i2c.c

```diff
@@ -132,11 +144,226 @@ void itc_iotbus_i2c_write_read_p(void)
 	TC_SUCCESS_RESULT();
 }
 
+/**
+* @testcase         itc_iotbus_i2c_set_frequency_address_p
+* @brief            sets the i2c frequency and address
+* @scenario         sets the i2c frequency and address
+* @apicovered       iotbus_i2c_set_frequency, iotbus_i2c_set_address
+* @precondition     initializes i2c_context
+* @postcondition    closes i2c_context
+*/
+void itc_iotbus_i2c_set_frequency_address_p(void)
+{
+	int ret = IOTBUS_ERROR_NONE;
+	uint8_t address = 0x8;
+	int freq_mode[] = { IOTBUS_I2C_STD, IOTBUS_I2C_FAST, IOTBUS_I2C_HIGH };
+	int num_modes = sizeof(freq_mode) / sizeof(freq_mode[0]);
+	int index = 0;
+	bool check = true;
+	iotbus_i2c_context_h h_i2c = iotbus_i2c_init(g_bus);
+	TC_ASSERT_NEQ("iotbus_i2c_init", h_i2c, NULL);
+
+	for (index = 0; index < num_modes; index++) {
+		ret = iotbus_i2c_set_frequency(h_i2c, freq_mode[index]);
+		if (ret != IOTBUS_ERROR_NONE) {
+			check = false;
+			SYSIO_ITC_PRINT("\nitc_iotbus_i2c_set_frequency_address_p: iotbus_i2c_set_frequency FAIL for frequency : %d\n", freq_mode[index]);
+			continue;
```

Comment:
Why continu? not break? Is there any difference between a failure and several failure?

Reply:
"continue" is used to check TC behavior for all the different frequencies.